src/Elastic.Markdown/Myst/Directives/Diagram/DiagramHttpClient.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+namespace Elastic.Markdown.Myst.Directives.Diagram;
+
+/// <summary>
+/// Shared HttpClient for diagram downloads to avoid resource exhaustion
+/// </summary>
+public static class DiagramHttpClient
```

Comment:
Create an `HttpClient` instance on `diagramregistry`. 

It will be easier for us to refactor later to use a single `IHttpClientFactory` for the lifetime of `docs-builder.